app/services/members/delete_service.rb

```diff
@@ -60,4 +60,13 @@ def cleanup_for_group(member)
       .new(member.principal, current_user: user, contract_class: EmptyContract)
       .call
   end
+
+  def default_contract_class
+    # We have different contracts for project roles and global roles
+    if model.project.present?
```

Comment:
There would also be the `#project_role?` helper method. Don't know if that it really better.

Reply:
Seems like a good fit here, added!